src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -71,75 +89,249 @@ const DotPlotPage = (props) => {
   const { experimentId } = props;
 
   const dispatch = useDispatch();
-  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+
+  const {
+    config,
+    plotData,
+    loading: plotDataLoading,
+    error: plotDataError,
+  } = useSelector((state) => state.componentConfig[plotUuid]) || {};
+
   const {
     fetching: genesFetching,
     data: highestDispersionGenes,
   } = useSelector((state) => state.genes.properties.views[plotUuid] || {});
-  const cellSets = useSelector((state) => state.cellSets);
+
+  const cellSets = useSelector(getCellSets());
   const {
     loading: cellSetsLoading,
     error: cellSetsError,
-    hierarchy,
+    hierarchy: cellSetHierarcy,
+    properties: cellSetProperties,
   } = cellSets;
 
-  useEffect(() => {
-    dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+  const [moreThanTwoGroups, setMoreThanTwoGroups] = useState(false);
 
-    if (hierarchy.length === 0) dispatch(loadCellSets(experimentId));
+  useEffect(() => {
+    if (!config) dispatch(loadPlotConfig(experimentId, plotUuid, plotType));
+    if (cellSetHierarcy.length === 0) dispatch(loadCellSets(experimentId));
   }, []);
 
+  const previousComparedConfig = useRef(null);
+  const getComparedConfig = (updatedConfig) => _.pick(
+    updatedConfig,
+    ['useMarkerGenes',
+      'nMarkerGenes',
+      'selectedGenes',
+      'selectedCellSet',
+      'selectedPoints'],
+  );
+
+  const hasGroupsToCompare = (baseCluster, filterCluster) => {
+    if (cellSetsLoading || !baseCluster || !filterCluster) return false;
+
+    // filterBy has the shape louvain/louvain-1
+    const [filterRootNode, filterKey] = filterCluster.split('/');
+
+    // If 'All" is chosen for the dropdown, there will always be representation from more than 1 group
+    if (filterRootNode === 'All') return true;
+
+    const filterClusterCellIds = Array.from(cellSetProperties[filterKey].cellIds);
+
+    const baseClusterParent = cellSetHierarcy.find((cellSet) => cellSet.key === baseCluster);
```

Comment:
Check if you can use the cellsets selectors here I think we can get hierarchy by key.

Reply:
Whoops, you've just uncovered a bug in React! I tried using a selector in the function (which is called inside a `useMemo`). Looking at the error, I don' think we can do this yet.

![image](https://user-images.githubusercontent.com/2862362/143239947-acd740be-5d05-43d9-85ef-530bb3f84150.png)

Reply:
no its not good to use hook inside a hook in general i think its one of the react hook rules, maybe you can do it on the top of the file as a variable.

Reply:
Also alternatively you can call the actual selector directly not through useSelector i guess?

Reply:
I don't think we can do that, because `getCellSetsSelector` requires `state`, which is passed by `useSelector`.

Reply:
I can move that line outside the function, but it makes it messy because we get the `baseClusterParent` as another global dependency from outside of the function and we lose the use of `baseCluster` as one of the keys of memoization.

Using the selectors is a good thing - the result is memoized. The result of the `hasGroups...` function is also memoized and the cellSetsHierarchy array is short, so I don't think the performance hit of not using the memoized selector would be a noticeable hit.

Reply:
This works,  but as I said, we lose the functionality of `baseCluster` as a key for useMemo, and we can't use `baseClusterParent` as a key as it is an object (whose reference changes everytime)

![image](https://user-images.githubusercontent.com/2862362/143244911-b7000c30-d948-4722-a2e0-cf37cd5c8cf4.png)

Reply:
So I'd still use the current approach. wdyt?

Reply:
You can pass the hierarchy and keys directly and  call `getCellSetsHierarchyByKeys`? As for the other things im not sure if it doesnt impact performance then maybe its good?

Reply:
Yup, that's what I did, as I show in the example above. However, I didn't really like it because : 
- It splits the code so that `baseClusterParent` away from where it's used, and causes `hasGroupsToCompare` to have another dependency on a global variable. 
- The data that is fetched by `getCellSetsHierarchyByKeys` already exists in `cellSets`, which is used in other parts of the code.
- Performance-wise, it's not that much of a difference. So I think the current code (without the selector) looks better as it has better legibility.

So I think the way it's currently coded looks fine.